packages/std/src/results/submessages.rs

```diff
@@ -5,9 +5,8 @@ use crate::Binary;
 
 use super::{CosmosMsg, Empty, Event};
 
-/// Use this to define when the contract gets a response callback.
-/// If you only need it for errors or success you can select just those in order
-/// to save gas.
+/// Specifies when a contract receives a callback response.
+/// Useful for optimizing gas usage and tailoring response handling.
```

Comment:
The "Useful for optimizing gas usage" part is confusing. The original wording was already somewhat confusing, so here's my understanding of it:
If you need to send a message and handle the error case, you could theoretically use `ReplyOn::Always` and check if it errored. But in order to save gas, you can use `ReplyOn::Error` instead. Same with `ReplyOn::Success` if you only need to handle the success case.
So, not the `ReplyOn` enum itself is useful for optimizing gas, but some of the cases will incur higher gas cost than others.